pkg/tagger/tagstore.go

```diff
@@ -131,31 +138,65 @@ func (e *entityTags) get(highCard bool) ([]string, []string) {
 	}
 
 	// Cache miss
-	var arrays [][]string
 	var sources []string
-	lowCardCount := 0
+	var tagSortList []tagSort
```

Comment:
To make lookup faster, what do you think about making `tagSortList` a map, with `tagName` as key, so the main case (tags are not duplicated) does not incur a full list traversal?

Reply:
That was my first approach but I had a hard time doing it, but I was able to iterate on this first draft, should be much cleaner now

Reply:
I'm mixed about the results https://github.com/DataDog/datadog-agent/pull/1383/commits/edd302dd7deae03399db40d586c25829f3718811

Might be a little more optimised but it's adding another loop level, WDYT?